misc/scripts/library-coverage/comment-pr.py

```diff
@@ -95,14 +110,37 @@ def comment_pr(repo, run_id):
 
         comment = comment_first_line + \
             "A recent commit removed the previously reported differences."
-    post_comment(comment, repo, pr_number)
+
+    if raw_comment_id is None:
+        post_initial_comment(comment, repo, pr_number)
+    else:
+        update_existing_comment(comment, repo, pr_number, raw_comment_id)
 
 
-def post_comment(comment, repo, pr_number):
+def post_initial_comment(comment, repo, pr_number):
     print(f"Posting comment to PR #{pr_number}")
     utils.subprocess_run(["gh", "pr", "comment", str(pr_number),
                          "--repo", repo, "--body", comment])
 
+def update_existing_comment(comment, repo, pr_number, raw_comment_id):
+    # Fetch existing comment, and validate:
+    # - comment belongs to the PR with number `pr_number`
+    # - comment starts with the expected prefix `comment_first_line`
+    # - comment author is github-actions[bot]
+    comment_author = "github-actions[bot]"
+    filter = f"select(.issue_url | endswith(\"{repo}/issues/{pr_number}\")) " + \
+        f"| select(.body | startswith(\"{comment_first_line}\")) " + \
+        f"| select(.user.login == \"{comment_author}\") " + \
+        "| .id"
+    comment_id = utils.subprocess_check_output(["gh", "api", f"repos/{repo}/issues/comments/{raw_comment_id}",
+                                                "--jq", filter]).strip()
```

Comment:
**Note/question:** This code uses `--jq` instead of `| jq --arg` like in the [qhelp version](https://github.com/github/codeql/blob/d56a9f0781306aa4eca59d34401f3bae562ce0c1/.github/workflows/post-pr-comment.yml#L56-L60). I used `--jq` instead because I could not easily pass a pipe to `subprocess.check_output`. However, I noticed that `@esbena's` [comment here](https://github.com/github/codeql/pull/8618#discussion_r842794405) says that using `--arg` is preferable. Let me know if this code needs to be changed to use `--arg` or if it is okay as-is.

Reply:
This looks okay. You're not using any values from the environment in the `jq` filter, instead you are crafting the filter using a Python template string to statically contain all the patterns you want, so I think no need for `--arg`.

The only other difference between `gh api | jq` and `gh api --jq` is that the latter behaves like `jq -r`, i.e. it strips the enclosing quotes in the resulting JSON string. That is ok here.